CMakeLists.txt

```diff
@@ -32,8 +32,20 @@ option(RE2C_BUILD_RE2RUST "Build re2rust executable (an alias for `re2c --lang r
 option(RE2C_BUILD_BENCHMARKS "Build benchmarks" OFF)
 option(RE2C_REGEN_BENCHMARKS "Regenerate C code for benchmarks" OFF)
 
+# check whether re2c is the root project
+if(CMAKE_SOURCE_DIR STREQUAL PROJECT_SOURCE_DIR)
+    set(RE2C_IS_ROOT_PROJECT TRUE)
+else()
+    set(RE2C_IS_ROOT_PROJECT FALSE)
+endif()
```

Comment:
I wonder if it can be rewritten more elegantly using generator expressions: https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#string-comparisons

Reply:
Likely not. I'm pretty certain generator expression don't work in conditionals.
They expand in a later step of the build process.
I'm not aware of a direct way to express this more consisely (that does not require a helper macro).

Reply:
Ok, thanks for confirming. In that case let's just move this block above line 14, so that options are grouped together.